### PR TITLE
Chore(client-sdk): Refactor SDK Dependencies

### PR DESCRIPTION
**Pull request by ajimae**
**What does this PR do**
Refactor all package dependencies into a client packages

**What are the tasks to be completed**
- [x] Merge `auth-middleware` package into `sdk-client` package 
- [x] Merge `http-middleware` into `sdk-client` package
- [x] Clean up entire import and compress all exports to a single entry point

**How can this be manually tested**
- Pull the remote develop branch and install all required dependencies
- Create an sdk-client (e.g [sdk-client](https://github.com/commercetools/commercetools-js-sdk-v2-training/blob/solutions/handson/client.js)) using the necessary middleware (auth and http middleware).
- Alternatively, you can switch directory into `sdk-client` package and run `yarn test`

**TODO:**
- [X] Remove `// @ts-nocheck` from the all files
- [X] Improve test coverage
- [X] Fix failing test on Circle-Ci

**Any background context**
This is in no way a finished feature, it is still a **work in progress**.

**Issue ID**
[Github Issue](https://github.com/commercetools/commercetools-sdk-typescript/issues/77)

**Screenshots**
None

### CODE REVIEW

File: packages/sdk-client/src/builder/ApiRootBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  }
+
+  build(): ApiRoot {
+    const { middlewares } = this;
+    if (this.authMiddleware) this.middlewares.push(this.authMiddleware);
+    if (this.httpMiddleware) this.middlewares.push(this.httpMiddleware);
+    if (this.userAgentMiddleware) this.middlewares.push(this.userAgentMiddleware);
+    if (this.correlationIdMiddleware) this.middlewares.push(this.correlationIdMiddleware);
+    if (this.loggerMiddleware) this.middlewares.push(this.loggerMiddleware);
+    if (this.queueMiddleware) this.middlewares.push(this.queueMiddleware);
```
Comment (jenschude) [inline packages/sdk-client/src/builder/ApiRootBuilder.ts:126]: the order here seems to be strange and should be clarified

imho it should be more like http, queue, logger, auth, userAgent, correlationId, everything else

File: packages/sdk-client/src/builder/ApiRootBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+      credentials
+    })
+      .withHttpMiddleware({
+        host: baseUri,
+        fetch: fetch,
+      })
+      .withLoggerMiddleware()
+  }
+
+  withAuthMiddleware(options: AuthMiddlewareOptions): ApiRootBuilder {
```
Comment (jenschude) [inline packages/sdk-client/src/builder/ApiRootBuilder.ts:50]: i would introduce two methods here

`withAuthMiddleware(authMiddleware: Middleware)`

and `withClientCredentialsFlow(options: AuthMiddlewareOptions)`

cause this method atm only supports client credentials flow. For other auth types a specific authMiddleware has to be added.

**Review by jenschude — CHANGES_REQUESTED** [review]

File: packages/sdk-client/src/builder/ApiRootBuilder.ts (hunk 1 above)
Comment (ajimae) [reply packages/sdk-client/src/builder/ApiRootBuilder.ts]: I have included other authentication middleware methods `withClientCredentialsFlow`, `withPasswordFlow`, `withAnonymousSessionFlow` `withRefreshTokenFlow` etc

Now, you can use any of the 5 client authentication method.

File: packages/sdk-client/src/builder/ApiRootBuilder.ts (hunk 2 above)
Comment (ajimae) [reply packages/sdk-client/src/builder/ApiRootBuilder.ts]: I have included other authentication middleware methods `withClientCredentialsFlow`, `withPasswordFlow`, `withAnonymousSessionFlow` `withRefreshTokenFlow` etc

Now, you can use any of the 5 client authentication method.

File: packages/sdk-client/src/builder/ClientBuilder.ts [hunk 3]
```diff
@@ -64,7 +64,7 @@ export default class ClientBuilder {
       .withLoggerMiddleware()
   }
 
-  withAuthMiddleware(authMiddleware: Middleware): ClientBuilder {
+  private withAuthMiddleware(authMiddleware: Middleware): ClientBuilder {
```
Comment (jenschude) [inline packages/sdk-client/src/builder/ClientBuilder.ts:67]: it's fine with this one as being public as it opens the possibility to add a custom AuthMiddleware

```suggestion
  withAuthMiddleware(authMiddleware: Middleware): ClientBuilder {
```

Comment (ajimae) [reply packages/sdk-client/src/builder/ClientBuilder.ts]: Done

**Review by jenschude — APPROVED** [review]

**Pull request merged**